etna/ciim/models.py

```diff
@@ -1,237 +1,77 @@
-from abc import ABC
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
 
 from django.conf import settings
+from django.utils.functional import classproperty
 
 from .client import KongClient
-from .exceptions import (
-    DoesNotExist,
-    InvalidQuery,
-    KongAPIError,
-    MultipleObjectsReturned,
-    UnsupportedSlice,
-)
-
-
-class SearchManager:
-    """A model.Manager/QuerySet-like object to query and fetch KongModel instances."""
-
-    def __init__(self, model):
-        self.model = model
-
-    def filter(self, **kwargs):
-        # Check we're making a meaningful query before contacting Kong
-        if any({k: v for k, v in kwargs.items() if v is None}):
-            raise InvalidQuery("Query to Kong must at least one clause")
-
-        return SearchQuery(self.model, kwargs)
-
-    def get(self, iaid=None, reference_number=None, expand=None):
-        return FetchQuery(self.model).get(iaid=iaid, reference_number=reference_number, expand=expand)
-
-    def get_multiple(self, iaid=None):
-        return FetchQuery(self.model).get_multiple(iaid=iaid)
+from .exceptions import DoesNotExist, MultipleObjectsReturned
 
 
-class ResultPending(Exception):
-    """Raised if result being requested that hasn't yet been fetched."""
+class APIModel(ABC):
+    """Model representation of API data.."""
 
+    @classmethod
+    @abstractmethod
+    def from_api_response(cls, response: dict) -> APIModel:
```

Comment:
Not sure why the model in our case the Record is handling the api reponse , I believe this is the APIManage duty

Reply:
Defining an abstract method on the model allows us to accommodate different model's needs.

Later down the line, we may re-implement `Image` as an `APIModel`. It's then the responsibility of the `Image` class to dictate how we go from response data to an `Image` instance.

Think of this method a bit like constructor overloading in C# 😄

Reply:
Not really the answer :)

Reply:
Related discussion https://github.com/nationalarchives/ds-wagtail/pull/234#pullrequestreview-888019163